pkg/db/jsonType.go

```diff
@@ -0,0 +1,60 @@
+package db
+
+import (
+	"database/sql"
+	"database/sql/driver"
+	"encoding/json"
+	"errors"
+
+	"github.com/justtrackio/gosoline/pkg/mdl"
+)
+
+var (
+	_                        driver.Valuer = JSONType[struct{}]{}
+	_                        sql.Scanner   = &JSONType[struct{}]{}
+	ErrJSONInvalidDriverType               = errors.New("invalid incoming driver type for json type")
```

Comment:
What is a driver type? Sounds like the type of a driver to connect to SQL. And why is it invalid? What would've been a valid type?